Thu Apr  3 17:50:27 CEST 2003	Arnaud Charlet	<charlet@gnat.com>

	Changes to ease distributing gtk+ executables with no gtk+ install:

	gtk/gtkimmodule.c: Handle more gracefully a missing immodule file

--- gtk+-2.2.1/gtk/gtkimmodule.c	2003/04/03 15:11:23	1.1
+++ gtk+-2.2.1/gtk/gtkimmodule.c	2003/04/03 15:12:35
@@ -279,9 +279,10 @@
   file = fopen (filename, "r");
   if (!file)
     {
-      g_warning ("Can not open Input Method module file '%s': %s",
-		 filename, g_strerror (errno));
-      /* We are leaking all kinds of memory here. */
+      /* The file does not exists, return immediately. */
+      g_string_free (line_buf, TRUE);
+      g_string_free (tmp_buf, TRUE);
+      g_free (filename);
       return;
     }
 
